File: Colored_bar_chart.R

categories <- c("A", "B", "C", "D")
values <- c(3, 7, -5, 2)  # Données avec variations positives et négatives
colors <- ifelse(values > 0, "green", "red")  # Couleurs selon la valeur

barplot(values, names.arg = categories, col = colors, main = "Diagramme en Barres Colorées",
        xlab = "Catégories", ylab = "Valeurs")

File: Colored_bar_chart/Colored_bar_chart.py
import matplotlib.pyplot as plt

categories = ['A', 'B', 'C', 'D']
values = [3, 7, -5, 2]  # Données avec des variations positives et négatives
colors = ['green' if v > 0 else 'red' for v in values]  # Attribuer une couleur selon la valeur

plt.bar(categories, values, color=colors)  # Création du diagramme
plt.title("Diagramme en Barres Colorées")
plt.xlabel("Catégories")
plt.ylabel("Valeurs")
plt.show()
